### PR TITLE
Improve :has() pseudo-class documentation

### LINKED ISSUE #19972: Improve the :has() page documentation
This issue is for tracking improvements to the [:has()](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) page.

Reference: https://github.com/mdn/content/issues/19572#issuecomment-1214645922

> - the syntax docs are take a <forgiving-relative-selector-list> but that is not defined anywhere in our docs. In summary all the "forgiving" bit means is that any of the comma separated selectors are invalid they just get skipped and the selector works on the valid ones.
>     - This is apparently different to normal list selectors, where if one is wrong the lot get dumped. Not clear if this applies to just when used in a query or in a style sheet.
>     - the "relative" bit means that the selectors are relative to the element or whatever that this has() was applied to.
> - The WG just added a new rule that has() pseudo class can't contain a pseudo class of the current types, but may take future types:
> https://github.com/w3c/csswg-drafts/issues/7463#issuecomment-1196971909
>     - So we may have to list the cases that can't be used.
>     - It is not clear whether FF already implements this restriction according to https://bugzilla.mozilla.org/show_bug.cgi?id=418039. I think we need some test code.
> 
> - The Description from/including "In earlier revisions of the CSS Selectors Level 4 specification, :has had a limitation that it couldn't be used within stylesheets. ..." should be removed. If it wasn't implemented in any browser it is now irrelevant.
> - The examples are too simplistic - they only show one item - NOT a list. We need a list, and we should also show that the case of an invalid selector is dropped. Ideally in a live example.

### PR DESCRIPTION
**Pull request by dipikabh**
### Description

This PR improves the documentation of the `:has()` page and updates the documentation of a few other pages where related terms appear. Here's a summary of these changes:

1. `:has()` pseudo-class page (https://developer.mozilla.org/en-US/docs/Web/CSS/:has)
- Improves the description and adds link to the definition of the term "forgiving selector list"
- Adds three examples to showcase different scenarios for using `:has()`

2. CSS selectors page (https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors)
- Improves the formatting on the page so that "Syntax" and "Examples" for the different type of selectors appear on newlines
- Adds the section "## Structure of a selector" to define the terms simple selectors, compound selectors, complex selectors, relative selectors, and selector lists

3. Pseudo-class page (https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) - includes minor language improvements

4. Selector list page (https://developer.mozilla.org/en-US/docs/Web/CSS/Selector_list)
- Explains the terms "forgiving selector list" and "forgiving relative selector list"

### Additional details

https://drafts.csswg.org/selectors/#relational

### Related issues and pull requests

Fixes #19972

### CODE REVIEW

File: files/en-us/web/css/_colon_has/index.md [hunk 1]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
 
-The **`:has()`** CSS [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents an element if any of the selectors passed as parameters (relative to the {{cssxref(":scope")}} of the given element) match at least one element.
+The functional **`:has()`** CSS [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents an element if any of the [relative selectors](/en-US/docs/Web/CSS/CSS_Selectors#relative_selector) that are passed as an argument match at least one element when anchored against this element. This pseudo-class presents a way of selecting a parent element or a previous sibling element with respect to a reference element by taking a [forgiving relative selector list](/en-US/docs/Web/CSS/Selector_list#forgiving_relative_selector_list) as an argument.
+
+```css
+/* Selects an h1 heading with a
+paragraph element that immediately follows
+the h1 and applies the style to h1 */
+h1:has(+ p) { margin-bottom: 0; }
+```
```
Comment (Josh-Cena) [inline files/en-us/web/css/_colon_has/index.md:23]: Is this intended to be replaced with an interactive example later? In general I think static examples should live in the "Examples" section.

**Review by estelle — COMMENTED** [review]
first edits....

File: files/en-us/web/css/_colon_has/index.md [hunk 2]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
 ## Syntax
 
 ```
 :has( <forgiving-relative-selector-list> )
 ```
 
-## Description
-
-The `:has()` pseudo-class takes a relative selector list as an argument.
+> **Note:** The `:has()` pseudo-class cannot be nested within another `:has()`. Also, pseudo-elements are not valid selectors within `:has()`. This is because many pseudo-elements exist conditionally based on the styling of their ancestors and allowing these to be queried by `:has()` can introduce cyclic querying.
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:31]:
```suggestion
The `:has()` pseudo-class cannot be nested within another `:has()`. Also, pseudo-elements are not valid selectors within `:has()`. This is because many pseudo-elements exist conditionally based on the styling of their ancestors and allowing these to be queried by `:has()` can introduce cyclic querying.

The relative selector list is [forgiving](https://drafts.csswg.org/selectors-4/#typedef-forgiving-selector-list). Normally in CSS when a selector in a selector list is invalid then the whole list is deemed invalid. When a selector in a `:has()` selector list fails to parse, the incorrect or unsupported selector will be ignored and the others used. Note that if the `:has()` itself is not supported in a browser, the entire selector block will fail (unless the :has() itself is in a forgiving selector list such as is found in [`:is()`](/en-US/docs/Web/CSS/:is) and [`:where()`](/en-US/docs/Web/CSS/:where).

While `:has()` cannot be nested in a `:has()`, as the selector list is forgiving, it will just be ignored. The selector will not fail. The same holds true for pseudo-elements.
```

File: files/en-us/web/css/_colon_has/index.md [hunk 3]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
 
 ## Examples
 
-### Matching \<a> elements that directly contain an \<img>
+### With the sibling combinator
+
+The `:has()` rule in the following example adjusts the spacing after `H1` headings if they are immediately followed by an `H2` heading.
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:37]:
```suggestion
The `:has()` style declaration in the following example adjusts the spacing after `H1` headings if they are immediately followed by an `H2` heading.
```

File: files/en-us/web/css/_colon_has/index.md [hunk 4]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+#### HTML
+
+```html
+<section>
+  <article>
+    <h1>Morning Times</h1>
+    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
+  </article>
+  <article>
+    <h1>Morning Times</h1>
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:48]: maybe call this afternoon times so you can reference and say the afternoon times heading is matched.

File: files/en-us/web/css/_colon_has/index.md [hunk 5]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+h1:has(+ h2) {
+  margin: 0 0 0.25rem 0;
+}
+```
+
+#### Result
+
+{{EmbedLiveSample('With_the_sibling_combinator', 600, 150)}}
+
+This example shows two similar texts side-by-side for comparison – the left one with an `H1` heading followed by a paragraph and the right one with an `H1` heading followed by an `H2` heading and then a paragraph. In the example on the right, `:has()` helps to select the `H1` element that is immediately followed by (indicated by [`+`](/en-US/docs/Web/CSS/Adjacent_sibling_combinator)) an `H2` element and the CSS rule reduces the spacing after such an `H1` element. Without the `:has()` pseudo-class, you cannot select the preceding sibling or parent element.
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:93]: you can actually select a preceding sibling if it is of the same type. I showed this with the fieldset example explanation

**Review by estelle — COMMENTED** [review]
my feedback is looking a little borked, so adding this (hope it doesn't fail), but still reviewing.

File: files/en-us/web/css/selector_list/index.md [hunk 6]
```diff
@@ -21,86 +21,73 @@ div {
...
-  font-family: sans-serif;
-}
+:is(h1, h2:maybe-unsupported, h3) { font-family: sans-serif; }
 ```
 
-This is because a single unsupported selector in a selector list invalidates the whole rule.
+### Forgiving relative selector list
 
-A way to remedy this is to use the {{CSSxRef(":is", ":is()")}} or {{CSSxRef(":where", ":where()")}} selectors, which accept a forgiving selector list. This will ignore invalid selectors in the list but accept those which are valid.
+A forgiving relative selector list is similar to the [forgiving selector list](#forgiving_selector_list); it just parses the list components as [relative selectors](/en-US/docs/Web/CSS/CSS_Selectors#relative_selector) rather than [complex selectors](/en-US/docs/Web/CSS/CSS_Selectors#complex_selector).
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:70]:
```suggestion
A forgiving relative selector list is similar to the [forgiving selector list](#forgiving_selector_list); it just parses the list components as [relative selectors](/en-US/docs/Web/CSS/CSS_Selectors#relative_selector), which begin with an explicit or implied combinator, rather than [complex selectors](/en-US/docs/Web/CSS/CSS_Selectors#complex_selector).
```

File: files/en-us/web/css/css_selectors/index.md [hunk 7]
```diff
@@ -12,90 +12,146 @@ spec-urls: https://drafts.csswg.org/selectors/
...
+    **Example:** `a#selected {...}`
+
+- Complex selector
+  - : A sequence of one or more [compound selectors](/en-US/docs/Web/CSS/CSS_Selectors#compound_selector) that are separated by [combinators](#combinators). A complex selector represents a set of simultaneous conditions on a set of elements. These conditions apply in the context of relationships described by the combinators. A given element is said to match a complex selector when the element matches compound selectors and the combinators between the compound selectors.
+
+    **Example:** `a#selected > .icon {...}`, `.box h2 + p {...}`
+
+- Relative selector
+
+  - : A complex selector beginning with a combinator. The [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator) is implied in case of a missing combinator.
```
Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:146]:
```suggestion
  - : A selector that represents an element relative to one or more anchor elements preceded by a combinator. Relative selectors that don't begin with an explicit [combinator](#combinators) have an implied [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator) assumed.
```

The spec does not explain anchor. Asking the author.... my edit likely isn't an improvement, but maybe.

File: files/en-us/web/css/css_selectors/index.md [hunk 8]
```diff
@@ -12,90 +12,146 @@ spec-urls: https://drafts.csswg.org/selectors/
...
   - : The `::` pseudo represent entities that are not included in HTML.
+
     **Example:** `p::first-line` will match the first line of all {{HTMLElement("p")}} elements.
 
+## Structure of a selector
+
+The term 'selector' can refer to one of the following:
+
+- Simple selector
+  - : An individual selector. A simple selector represents a single condition on an element. A given element is said to match a simple selector when that simple selector accurately describes the element. All [basic selectors](#basic_selectors) and [pseudo-classes](#pseudo-classes_and_pseudo-elements) are simple selectors.
```
Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:130]:
```suggestion
  - : A selector with a single component such as a single id selector, type selector, classesselector, including a single pseudo-class, attribute, or pseudo-element, not used in combination with or containing any other selector component or combinator.  
```
we need to bring all of these descriptions to a 3rd to 8th grade level. This sounds spec-y (same for compound and complex)

File: files/en-us/web/css/_colon_has/index.md [hunk 9]
```diff
@@ -53,5 +270,8 @@ h1:has(+ p)
 
 ## See also
 
+- [`:is()`](/en-US/docs/Web/CSS/:is), [`:not()`](/en-US/docs/Web/CSS/:not)
+- [CSS selectors](/en-US/docs/Web/CSS/CSS_Selectors)
+- [Selector list](/en-US/docs/Web/CSS/Selector_list)
 - [Locating DOM elements using selectors](/en-US/docs/Web/API/Document_object_model/Locating_DOM_elements_using_selectors)
-- [`:scope`](/en-US/docs/Web/CSS/:scope)
+- For more inspiring ideas, see [Using :has() as a CSS parent selector and much more](https://webkit.org/blog/13096/css-has-pseudo-class/#using-has-with-the-child-combinator).
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:277]:
```suggestion
```

File: files/en-us/web/css/_colon_has/index.md [hunk 10]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+section:not(:has(h1, h2, h3, h4, h5, h6)) {...}
+```
+
+However, if you swap the order of the two pseudo-classes as shown below:
+
+```css
+section:has(:not(h1, h2, h3, h4, h5, h6)) {...}
+```
+
+The `:not()` pseudo-class will select any element that's not a heading element, such as `<p>`, `<a>`, or `<button>` and pass that as an argument to `:has()`, which in turn, will select any `<section>` element that contains those other non-heading elements. This might eventually end up selecting even those `<section>` elements that do have a heading element along with other non-heading elements.
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:261]:
```suggestion
The `:not()` pseudo-class will select any element that's not a heading element, such as `<p>`, `<a>`, or `<button>` and pass that as an argument to `:has()`, which in turn, will select any `<section>` element that contains those other non-heading elements. This matches every section that contains a single element that is not a heading, meaning it will match a section that contains both paragraphs and headings. For this reason, it is recommended to always use explicit, specific combinators in all components `:has()` within the relational selector list parameter.
```

File: files/en-us/web/css/_colon_has/index.md [hunk 11]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+}
+```
+
+#### Result
+
+{{EmbedLiveSample('With_the_:not()_pseudo-class', 600, 300)}}
+
+- The first `:has()` pseudo-class selects any `<fieldset>` element adjacent to another `<fieldset>` element and applies a bottom spacing to the preceding `<fieldset>` element. This selects the first and second `<fieldset>` elements, and the spacing is applied after them.
+
+- The second `:has()` pseudo-class in combination with the [`:not()`](/en-US/docs/Web/CSS/:not) pseudo-class selects any `<fieldset>` element with the `required` class containing an input radio or checkbox that hasn't been toggled to an [on state](/en-US/docs/Web/CSS/:checked). This selects the first and second `<fieldset>` elements, and applies the pink background to the `<fieldset>` elements.
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:243]: let's add a 1px red outline as well, so color is not the only thing changed. Outline, not border, so the box model isn't altered.

File: files/en-us/web/css/_colon_has/index.md [hunk 12]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+    <legend>Which activity are you interested in?</legend>
+    <input type="radio" name="activity" id="activity1">
+    <label for="activity1">Running</label> 
+    <input type="radio" name="activity" id="activity2">
+    <label for="activity2">Rock climbing</label>
+    <input type="radio" name="activity" id="activity3">
+    <label for="activity3">Swimming</label>
+  </fieldset>
+
+  <fieldset class="required">
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:181]: if this is required, it has to be required: the screen reader user has no way of knowing it's required. And, the browser doesn't know either.

File: files/en-us/web/css/_colon_has/index.md [hunk 13]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+fieldset:has(+ fieldset) {
+  margin-bottom: 1em;
+}
+
+fieldset.required:not(:has(input:checked)) {
+  background: pink;
+}
+
+fieldset.required:not(:has(input:checked)) legend::after {
+  content: " ✳︎";
```
Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:232]:
```suggestion
  content: " (required)";
```

Comment (estelle) [reply files/en-us/web/css/_colon_has/index.md:232]: i am not sure if adding content is ok with CSS, or if we should also had an aria attribute

Comment (estelle) [reply files/en-us/web/css/_colon_has/index.md:232]: we could teach :has() and a11y at the same time.

```html
<legend>Label here <span aria-label="required">*</span></legend>
```

```css 
fieldset:has( > legend > [aria-label="required"])
```

That shows ho to correctly require a checkbox group and how to use the child selector in a has.

Comment (estelle) [inline files/en-us/web/css/_colon_has/index.md:232]: This is not best practices. Instead use:

```html
<fieldset>
   <legend>What days work for you? <span aria-label="required">*</span></legend>
```

and then in CSS you can teach:

```css
fieldset:has( > legend > [aria-label="required"])
/* or */
legend:has( > [aria-label="required"])
```

also, since fieldsets can be nested, this example, as is, may or may not be robust. Plus you don't want to toggle the generated content based on the state of an input. That would be weird.

File: files/en-us/web/css/_colon_has/index.md (hunk 1 above)
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:23]: Hi @Josh-Cena, here I have followed the [template for selector pages](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_selector_page_template), where it mentions using "/* Insert code block showing common use cases */" right before the "Syntax" section.

For example, see these pages [`:first`](https://developer.mozilla.org/en-US/docs/Web/CSS/:first) and [`:dir()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:dir). 

But then there's also this format in [`:hover`](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover). Maybe when we add an interactive example to `:has()`, this static example can be moved down to "Examples".

File: files/en-us/web/css/_colon_has/index.md (hunk 2 above)
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:31]: - Any reason for not having that text as "Note:"?
- Actually, the first paragraph on the page links to the definition of the 'forgiving selector list':
"This pseudo-class presents a way of selecting a parent element or a previous sibling element with respect to a reference element by taking a [forgiving relative selector list](https://pr20889.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/Selector_list#forgiving_relative_selector_list) as an argument."

In this PR, I also submitted the updated 'Selector list' page to update those definitions. Do we need to explain it here again?

File: files/en-us/web/css/css_selectors/index.md [hunk 14]
```diff
@@ -12,90 +12,146 @@ spec-urls: https://drafts.csswg.org/selectors/
...
+    **Example:** `a#selected > .icon {...}`, `.box h2 + p {...}`
+
+- Relative selector
+
+  - : A complex selector beginning with a combinator. The [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator) is implied in case of a missing combinator.
+
+    **Example:** `+ div#topic > #reference {...}`
+
+- [Selector list](/en-US/docs/Web/CSS/Selector_list)
+  - : A comma-separated list of [simple](/en-US/docs/Web/CSS/CSS_Selectors#simple_selector), [compound](/en-US/docs/Web/CSS/CSS_Selectors#compound_selector), or [complex](/en-US/docs/Web/CSS/CSS_Selectors#complex_selector) selectors. If the constituent selector type of a selector list is important but unspecified, it is called a _complex selector list_. A given element is said to match a selector list when the element matches any (at least one) of the selectors in that selector list. Read more about when a selector list is deemed [invalid](/en-US/docs/Web/CSS/Selector_list#invalid_selector_list) and how to construct a [forgiving selector list](/en-US/docs/Web/CSS/Selector_list#forgiving_selector_list).
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/css_selectors/index.md:151]: Spec wording. Would you know how to reword it to make it less ambiguous?

File: files/en-us/web/css/css_selectors/index.md (hunk 7 above)
Comment (dipikabh) [reply files/en-us/web/css/css_selectors/index.md:146]: I added a line about anchor elements in the pseudo-classes page

File: files/en-us/web/css/_colon_has/index.md [hunk 15]
```diff
@@ -13,36 +13,253 @@ browser-compat: css.selectors.has
...
+    <label for="day1">Monday</label> 
+    <input type="checkbox" id="day2">
+    <label for="day2">Wednesday</label>
+    <input type="checkbox" id="day3">
+    <label for="day3">Friday</label>
+  </fieldset>
+
+  <fieldset>
+    <legend>Join our newsletter</legend>
+    <input type="radio" name="newsletter" id="newsweekly">
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:193]: I am removing this form example for now. It seems we need a better example that correctly reflects the importance of `:has()` while also using other coding best practices. We can work on an example offline and then add it to the docs.

File: files/en-us/web/css/_colon_has/index.md (hunk 5 above)
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:93]: How would you recommend to fix this? Use `<p>` instead of `<h2>`?

File: files/en-us/web/css/_colon_has/index.md (hunk 2 above)
Comment (estelle) [reply files/en-us/web/css/_colon_has/index.md:31]: The one note became a 3 paragraph explanation, so didn't want to notify it. As the list is forgiving including `:has()` or `::pseudoelement` will not break anything, with the why in the next two paragraphs, I wanted to keep all three on the same level. 

While it does link to the forgiving selector list and to the relative selector, many people don't know what these terms mean. We don't want 100% of people to need to leave the page to understand its contents. Many of the terms can be linked to; but I strongly feel these need to be included in this document.

File: files/en-us/web/css/selector_list/index.md [hunk 16]
```diff
@@ -21,41 +21,21 @@ div {
...
-h1, h2, h3, h4, h5, h6 {
-  font-family: helvetica;
-}
-```
-
-### Multi Line Grouping
+## Description
 
-Grouping selectors in a multiple lines using a comma-separated lists.
+To reduce the size of style sheets, one can group selectors in a comma-separated list.
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:32]:
```suggestion
When applying styles to elements matching different criteria, group the selectors in a comma-separated list. This improves consistency while reducing the size of style sheets.
```

File: files/en-us/web/css/selector_list/index.md [hunk 17]
```diff
@@ -70,16 +50,18 @@ h3 {
...
-h1,
-h2:maybe-unsupported,
-h3 {
+h1, h2:maybe-unsupported, h3 {
   font-family: sans-serif;
 }
 ```
 
-This is because a single unsupported selector in a selector list invalidates the whole rule.
+They are not equivalent. In the first rule set, styles will be applied on the `h1` and `h3` elements, but the `h2` rule will not be parsed. However, in the second rule set, because one selector in the list is invalid, the entire rule will not be parsed and no style will be applied to the `h1` and `h3` elements as well.
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:58]:
```suggestion
They are not equivalent. In the first rule set, styles will be applied on the `h1` and `h3` elements, but the `h2` rule will not be parsed. However, in the second rule set, because one selector in the list is invalid, the entire rule will not be parsed and no style will be applied to the `h1` and `h3` elements as well; the entire style block is ignored.
```

File: files/en-us/web/css/selector_list/index.md [hunk 18]
```diff
@@ -99,6 +81,34 @@ h3 {
 }
 ```
 
+### Forgiving relative selector list
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:84]:
```suggestion
The difference between the two is that the specificity of `:is()` is its most specific argument whereas the `:where()` selector and the forging selector list parameter do not add any specificity weight. 

### Forgiving relative selector list
```

File: files/en-us/web/css/selector_list/index.md [hunk 19]
```diff
@@ -99,6 +81,34 @@ h3 {
...
+
+A forgiving relative selector list is similar to the [forgiving selector list](#forgiving_selector_list); it just parses the list components as [relative selectors](/en-US/docs/Web/CSS/CSS_Selectors#relative_selector), which begin with an explicit or implied combinator, rather than [complex selectors](/en-US/docs/Web/CSS/CSS_Selectors#complex_selector).
+
+## Examples
+
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:89]: what is the purpose of these two examples?

File: files/en-us/web/css/selector_list/index.md [hunk 20]
```diff
@@ -109,4 +119,6 @@ h3 {
 
 ## See also
 
-- The {{CSSxRef(":is", ":is()")}} {{Experimental_Inline}} and {{CSSxRef(":where", ":where()")}} {{Experimental_Inline}} pseudo-classes, which accept a forgiving selector list.
+- The {{CSSxRef(":is", ":is()")}} and {{CSSxRef(":where", ":where()")}} pseudo-classes accept a forgiving selector list.
+- [Forgiving selector list](https://drafts.csswg.org/selectors/#typedef-forgiving-selector-list)
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:123]: Link the forgiving selector list from the definition to the spec if you want, but don't add it in the see also as it doesn't provide any more detail than the MDN description. 
```suggestion
```

**Review by estelle — CHANGES_REQUESTED** [review]

File: files/en-us/web/css/_colon_has/index.md (hunk 2 above)
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:31]: I feel if the same concept is explained on different pages, it will lead to maintenance issues. The explanation of the behavior and examples can live on one page and others can link to it. The fact that folks don't know about these terms is relevant only now and but hopefully, more people will understand these terms in the future, in which case it will make sense for the concepts to live on a single page.

Currently, [`:where()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:where#forgiving_selector_parsing) and [`:is()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is#forgiving_selector_parsing) each have a section on "Forgiving selector parsing", perhaps because it was a new concept at the time. But that has now resulted in a repition of the same content, which leads to maintenance issues - if it is in one place, only one place would need to be updated in the future if there's any change in behavior or examples need to be updated. I wanted to update `:is()` and `:where()` via a separate PR.

File: files/en-us/web/css/_colon_has/index.md [hunk 21]
```diff
@@ -53,5 +169,7 @@ h1:has(+ p)
 
 ## See also
 
+- [`:is()`](/en-US/docs/Web/CSS/:is), [`:not()`](/en-US/docs/Web/CSS/:not)
+- [CSS selectors](/en-US/docs/Web/CSS/CSS_Selectors)
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/_colon_has/index.md:173]: I used the learn link. Thanks!

File: files/en-us/web/css/_colon_has/index.md (hunk 5 above)
Comment (estelle) [reply files/en-us/web/css/_colon_has/index.md]: fix in comment below.

File: files/en-us/web/css/css_selectors/index.md [hunk 22]
```diff
@@ -13,90 +13,146 @@ spec-urls: https://drafts.csswg.org/selectors/
...
+
+- Compound selector
+  - : A sequence of [simple selectors](#simple_selector) that are not separated by a [combinator](#combinators). A compound selector represents a set of simultaneous conditions on a single element. A given element is said to match a compound selector when the element matches all the simple selectors in the compound selector.
+
+    In a compound selector, a [type selector](/en-US/docs/Web/CSS/Type_selectors) or a [universal selector](/en-US/docs/Web/CSS/Universal_selectors) in a compound selector must come first in the sequence of selectors. Only one type selector or a universal selector is allowed in the sequence. Since whitespace represents the [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator), no whitespace is allowed between the simple selectors in a compound selector.
```
Comment (estelle) [inline files/en-us/web/css/css_selectors/index.md:136]:
```suggestion
    In a compound selector, the [type selector](/en-US/docs/Web/CSS/Type_selectors) or [universal selector](/en-US/docs/Web/CSS/Universal_selectors) must come first in the sequence of selectors. Only one type selector or the universal selector is allowed in the sequence. Since whitespace represents the [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator), no whitespace is allowed between the simple selectors in a compound selector.
```

File: files/en-us/web/css/selector_list/index.md (hunk 19 above)
Comment (estelle) [reply files/en-us/web/css/selector_list/index.md:92]: These should come first, not last, as they are used in all the other examples and explanations. Also, they are part of the definition, so not in an examples section, but rather included under their own heading with a brief sentence, and then a statement saying either is ok, with some company guidelines and linters preferring one over the other, so basically pick one, and stick with it.

File: files/en-us/web/css/selector_list/index.md [hunk 23]
```diff
@@ -12,57 +12,86 @@ browser-compat: css.selectors.list
 
 {{CSSRef}}
 
-The CSS **selector list** (`,`) selects all the matching nodes.
+The CSS **selector list** (`,`) selects all the matching nodes. A selector list is a comma-separated list of selectors.
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:15]:
```suggestion
The CSS **selector list** (`,`) selects all the matching nodes. A selector list is a comma-separated list of selectors.
```

**Review by estelle — COMMENTED** [review]
I made a large commit to files/en-us/web/css/selector_list/, reordering the page. I tried to do a separate PR, but lost my changes, rewrote it and tried again, but got a 404, so committed them. I put expanded the short definition, added a selector list in a functional selector in the syntax, and put the examples of simple lists before the explanation of forgiving since those example use lists.  I also added an example to the relative selector list, and added both has and not to the forgiving list as parameters line in the see also section.

I also added the four selectors to the pseudo-class list page: https://github.com/mdn/content/pull/21210

File: files/en-us/web/css/selector_list/index.md [hunk 24]
```diff
@@ -12,57 +12,86 @@ browser-compat: css.selectors.list
...
 .content,
-article {
+article,
+h1 + p {
   font-size: 1.1em;
 }
 ```
 
-### Selector list invalidation
+## Valid and invalid selector lists
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:80]:
```suggestion
## Invalid and forgiving selector lists
```

File: files/en-us/web/css/selector_list/index.md [hunk 25]
```diff
@@ -71,16 +100,18 @@ h3 {
 ```
 
 ```css
-h1,
-h2:maybe-unsupported,
```
Comment (estelle) [inline files/en-us/web/css/selector_list/index.md:75]: changed from unsupported to invalid because some selectors are not supported but are not invalid, such as FF neither supporting nor failing on `-webkit` prefixed pseudo-elements. So, while neight `::foobar` nor `::-webkit-foobar` is supported in FF, FF only considers the former invalid and will ignore the whole declaration.

**Comment by dipikabh** [comment]
Hi @estelle, we can probably land this now and make more iterations in future PRs. Thank you!

**Review by estelle — APPROVED** [review]
